docs/content/Schema/Fundamentals/Concepts.md

```diff
@@ -0,0 +1,303 @@
+dimensions allow granulating the result set for queries sent to Cube.js.
+
+We can add the necessary time dimensions to both schemas as follows:
+
+```javascript
+cube('Orders', {
+
+  ...,
+
+  dimensions: {
+
+    ...,
+
+    createdAt: {
+      sql: `created_at`,
+      type: `time`
+    },
+
+    completedAt: {
+      sql: `completed_at`,
+      type: `time`,
+    },
+  },
+});
+```
+
+```javascript
+cube('LineItems', {
+
+  ...,
+
+  dimensions: {
+
+    ...,
+
+    createdAt: {
+      sql: `created_at`,
+      type: `time`
+    },
+  },
+});
+```
+
+Time dimensions are essential to enabling performance boosts such as
+[partitioned pre-aggregations][ref-caching-use-preaggs-partition-time] and
+[incremental refreshes][ref-tutorial-incremental-preagg].
+
+## Measures
+
+Measures represent the properties of a **set of data points** in the cube. To
+add a measure called `count` to our `Orders` cube, for example, we can do the
+following:
+
+```javascript
+cube('Orders', {
+
+  ...,
+
+  measures: {
+    count: {
+      type: `count`,
+    },
+  },
+});
+```
+
+In our `LineItems` cube, we can We could also create a measure to sum up the
+total value of line items sold:
+
+```javascript
+cube('LineItems', {
+
+  ...,
+
+  measures: {
+    total: {
+      sql: `price`,
+      type: `sum`,
+    },
+  },
+})
+```
+
+Measures can be of different types, and you can find them all
+[here][ref-schema-measure-types].
+
+## Joins
+
+Joins define the relationships between cubes, which then allows accessing and
+comparing properties from two or more cubes at the same time. In Cube.js, all
+joins are `LEFT JOIN`s by default.
+
+<!-- prettier-ignore-start -->
+[[info | ]]
+| An `INNER JOIN` can be replicated with Cube.js; when making a Cube.js query,
+| add a filter for `IS NOT NULL` on the required column.
+<!-- prettier-ignore-end -->
+
+In the following example, we are left-joining the `LineItems` cube onto our
+`Orders` cube:
+
+```javascript
+cube('Orders', {
+
+  ...,
+
+  joins: {
+    LineItems: {
+      relationship: `belongsTo`,
+      sql: `${CUBE}.id = ${LineItems}.order_id`,
+    },
+  },
+});
+```
+
+There are three kinds of join relationships:
+
+- `belongsTo`
+- `hasOne`
+- `hasMany`
+
+More information can be found in the [Joins reference
+documentation][ref-schema-ref-joins-types].
+
+## Segments
+
+Segments are filters that are predefined in the schema instead of [a Cube.js
+query][ref-backend-query-filters]. They allow simplifying Cube.js queries and
+make it easy to re-use common filters across a variety of queries.
+
+To add a segment which limits results to completed orders, we can do the
+following:
+
+```javascript
+cube('Orders', {
+  ...,
+  segments: {
+    onlyCompleted: {
+      sql: `${CUBE}.status = 'completed'`
+    },
+  },
+});
+```
+
+## Pre-Aggregations
+
+Pre-aggregations are a powerful way of caching frequently-used, expensive
+queries and keeping the cache up-to-date on a periodic basis. Within a data
```

Comment:
I assumed that we want users to use pre-aggregations for **all** queries rather than just **frequently-used, expensive** ones because otherwise we cannot provide any guarantees w.r.t. performance. I think it would be great to come to a shared understanding here. Cc @hassankhan @keydunov @ryanpei